dbms/src/IO/WriteHelpers.h

```diff
@@ -173,7 +174,7 @@ inline void writeString(const StringRef & ref, WriteBuffer & buf)
  *  - it is assumed that string is in UTF-8, the invalid UTF-8 is not processed
  *  - all other non-ASCII characters remain as is
  */
-inline void writeJSONString(const char * begin, const char * end, WriteBuffer & buf)
+inline void writeJSONString(const char * begin, const char * end, WriteBuffer & buf, const FormatSettings & settings = FormatSettings())
```

Comment:
Try to remove default whenever possible.

Reply:
Not in every place the settings are available, e.g. `FileChecker::save` method doesn't have settings instance available.
How should we approach this case?

Reply:
Just create default FormatSettings before call to this function.
If the function is called in the loop - move FormatSettings out of the loop, just in case.